test/test_nn.py

```diff
@@ -8956,6 +8956,37 @@ def __call__(self, input):
     input_size=(4, 16),
     fullname='AdaptiveLogSoftmax'))
 
+class _LayerNormWithNonDefaultWeight(nn.LayerNorm):
+    def __init__(self, normalized_shape, eps, elementwise_affine):
+        super().__init__(normalized_shape, eps, elementwise_affine)
+        self.weight.data = 0.1 * torch.ones_like(self.weight.data)
+
+class TestLayerNormWithNonDefaultWeight(NewModuleTest):
+    def __init__(self):
+        super().__init__(
+            constructor=lambda: _LayerNormWithNonDefaultWeight((16,), 1e-5, True),
+            input_size=(4, 4, 16),
+            check_gradgrad=True,
+            fullname='LayerNormWithNonDefaultWeight')
+
+    def compare_two_gradients(self, test_case, module, input):
+        num_threads = torch.get_num_threads()
+        torch.set_num_threads(1)
+
+        grads1 = torch.autograd.grad(module(input).sum(), input, create_graph=False)[0]
+        grads2 = torch.autograd.grad(module(input).sum(), input, create_graph=True)[0]
+
+        atol = 1e-5
+        rtol = 1e-3
+        test_case.assertTrue(torch.allclose(grads1, grads2, rtol, atol))
+
+        torch.set_num_threads(num_threads)
+
+    def _do_test(self, test_case, module, input):
+        self.compare_two_gradients(test_case, module, input)
+        super()._do_test(test_case, module, input)
+
+add_test(TestLayerNormWithNonDefaultWeight())
```

Comment:
If you just want to add a simple test outside of the generic test framework, you can just add a new method to the `TestNN` class above and it will just run.

Reply:
Made a test class inherited from `TestCase`

Reply:
We usually avoid new classes as, as you can see just below, we have custom setup code that is simpler to handle if you have a shared class.
Would it be possible to just move this test in the `TestNN` class that is just below?